src/tests/notify.bats

```diff
@@ -55,52 +58,104 @@ setup() {
-    [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
+    [ "$status" -eq 0 ]
 }
 
 @test "7: FilterBy - string" {
     CIRCLE_BRANCH="master"
     run FilterBy "$SLACK_PARAM_BRANCHPATTERN" "$CIRCLE_BRANCH"
     echo "Error output debug: $output"
-    [ "$output" == "" ] # "master" is in the list: No output error
-    [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
+    [ "$status" -eq 0 ]
 }
 
 @test "8: FilterBy - regex numbers" {
     CIRCLE_BRANCH="pr-123"
     run FilterBy "$SLACK_PARAM_BRANCHPATTERN" "$CIRCLE_BRANCH"
     echo "Error output debug: $output"
-    [ "$output" == "" ] # "pr-[0-9]+" should match: No output error
-    [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
+    [ "$status" -eq 0 ]
 }
 
 @test "9: FilterBy - non-match" {
     CIRCLE_BRANCH="x"
     run FilterBy "$SLACK_PARAM_BRANCHPATTERN" "$CIRCLE_BRANCH"
     echo "Error output debug: $output"
-    [[ "$output" =~ "NO SLACK ALERT" ]] # "x" is not included in the filter. Error message expected.
-    [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
+    [ "$status" -eq 1 ]
 }
 
 @test "10: FilterBy - no partial-match" {
     CIRCLE_BRANCH="pr-"
     run FilterBy "$SLACK_PARAM_BRANCHPATTERN" "$CIRCLE_BRANCH"
     echo "Error output debug: $output"
-    [[ "$output" =~ "NO SLACK ALERT" ]] # Filter dictates that numbers should be included. Error message expected.
+    [ "$status" -eq 1 ]
+}
+
+@test "ShouldPost - FilterBy branch passes" {
+    CIRCLE_BRANCH="master"
+    run ShouldPost
+    echo "Error output debug: $output"
+    [[ "$output" = "Posting status" ]]
+    [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
+}
+
+@test "ShouldPost - FilterBy branch fails" {
+    CIRCLE_BRANCH="x"
+    run ShouldPost
+    echo "Error output debug: $output"
+    [[ "$output" =~ "NO SLACK ALERT" ]]
+    [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
+}
+
+@test "ShouldPost - CIRCLE_BRANCH is empty" {
+    unset CIRCLE_BRANCH
+    run ShouldPost
+    echo "Error output debug: $output"
+    [[ "$output" =~ "NO SLACK ALERT" ]]
     [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
 }
 
-@test "11: FilterBy - SLACK_PARAM_BRANCHPATTERN is empty" {
+@test "ShouldPost - SLACK_PARAM_BRANCHPATTERN is empty" {
     unset SLACK_PARAM_BRANCHPATTERN
     CIRCLE_BRANCH="master"
-    run FilterBy "$SLACK_PARAM_BRANCHPATTERN" "$CIRCLE_BRANCH"
+    run ShouldPost
     echo "Error output debug: $output"
+    [[ "$output" =~ "NO SLACK ALERT" ]]
     [ "$status" -eq 0 ] # In any case, this should return a 0 exit as to not block a build/deployment.
 }
 
-@test "12: FilterBy - CIRCLE_BRANCH is empty" {
```

Comment:
the test numbers here are actually redundant; `bats` automatically prepends each test with the number, so currently the test output looks like
```
12 12: Filter By - CIRCLE_BRANCH is empty
```
Since I'm adding tests here, I figured I might as well start deduplicating the test numbers, but I can add them back for now, if you want